backend/views/syson-diagram-common-view/src/main/java/org/eclipse/syson/diagram/common/view/services/ViewToolService.java

```diff
@@ -680,15 +680,7 @@ public RequirementUsage becomeObjectiveRequirement(RequirementUsage requirement,
     }
 
     public PartUsage addAsNestedPart(PartDefinition partDefinition, PartUsage partUsage) {
-        var eContainer = partUsage.eContainer();
-        if (eContainer instanceof FeatureMembership featureMembership) {
-            partDefinition.getOwnedRelationship().add(featureMembership);
-        } else if (eContainer instanceof OwningMembership owningMembership) {
-            var newFeatureMembership = SysmlFactory.eINSTANCE.createFeatureMembership();
-            newFeatureMembership.getOwnedRelatedElement().add(partUsage);
-            partDefinition.getOwnedRelationship().add(newFeatureMembership);
-            this.deleteService.deleteFromModel(owningMembership);
-        }
+        this.changeOwner(partUsage, partDefinition);
```

Comment:
A unit test for `ViewToolService#addAsNestedPart` could be great for non-regression. Could you add one please?

Reply:
Done, in the end I merged `changeOwner` and `moveMembership`, which were doing the same thing.
The `moveMembership` is now in `UtilService` to make it reusable **and** unit-testable (`ViewToolService` requires too many parameters to be easily unit-testable).
There is now a new class `UtilServiceTest` with tests related to this method. A discussion is needed on what we can unit test, and what should be refactored to ease testing.